### PR TITLE
chore(apidocs): Add documentation for ProjectEventsEndpoint

### PR DESCRIPTION
**Pull request by malwilley**
Ref https://github.com/getsentry/getsentry/issues/13865

Depends on https://github.com/getsentry/sentry/pull/78133

Update these docs: https://docs.sentry.io/api/events/list-a-projects-error-events/

New docs page looks like this:

CleanShot 2024-09-25 at 12 47 44@2x

### CODE REVIEW

File: src/sentry/issues/endpoints/project_events.py [hunk 1]
```diff
@@ -31,26 +38,42 @@ class ProjectEventsEndpoint(ProjectEndpoint):
...
+                description="Return events in pseudo-random order. This is deterministic, same query will return the same events in the same order.",
+                required=False,
+                type=bool,
+                location="query",
+                default=False,
+            ),
+        ],
+        responses={
+            200: inline_sentry_response_serializer(
+                "ProjectEventsResponseDict", list[SimpleEventSerializerResponse]
```
Comment (sentaur-athena) [inline src/sentry/issues/endpoints/project_events.py:66]: SimpleEventSerializerResponse has SimpleEventTag which has a param query. Now query looks like it's optional but that's not how documentation reads it so it thinks it's required. You should use totality to say it's actually optional.
Here's more explained version in docs: https://develop.sentry.dev/application/api/public/#success-responses
Ping us if you have questions.

File: src/sentry/issues/endpoints/project_events.py [hunk 2]
```diff
@@ -31,26 +38,42 @@ class ProjectEventsEndpoint(ProjectEndpoint):
...
+                name="full",
+                description="If this is set to true then the event payload will include the full event body, including the stacktrace. Set to 1 to enable.",
+                required=False,
+                type=bool,
+                location="query",
+                default=False,
+            ),
+            OpenApiParameter(
+                name="sample",
+                description="Return events in pseudo-random order. This is deterministic, same query will return the same events in the same order.",
```
Comment (schew2381) [inline src/sentry/issues/endpoints/project_events.py:57]: nit: slightly better wording?
```suggestion
                description="Return events in pseudo-random order. This is deterministic so an identical query will always return the same events in the same order.",
```

Comment (malwilley) [reply src/sentry/issues/endpoints/project_events.py]: This is better!

File: api-docs/paths/events/project-events.json [hunk 3]
```diff
@@ -1,108 +0,0 @@
-{
```
Comment (sentaur-athena) [inline api-docs/paths/events/project-events.json:1]: Feels good to see all this red 💆‍♀️

File: src/sentry/issues/endpoints/project_events.py [hunk 4]
```diff
@@ -31,26 +38,42 @@ class ProjectEventsEndpoint(ProjectEndpoint):
...
 
+    @extend_schema(
+        operation_id="List a Project's Error Events",
+        parameters=[
+            GlobalParams.ORG_ID_OR_SLUG,
+            GlobalParams.PROJECT_ID_OR_SLUG,
+            CursorQueryParam,
+            OpenApiParameter(
+                name="full",
+                description="If this is set to true then the event payload will include the full event body, including the stacktrace. Set to 1 to enable.",
```
Comment (sentaur-athena) [inline src/sentry/issues/endpoints/project_events.py:49]: Visible in docs

File: src/sentry/issues/endpoints/project_events.py [hunk 5]
```diff
@@ -31,26 +38,42 @@ class ProjectEventsEndpoint(ProjectEndpoint):
...
+                required=False,
+                type=bool,
+                location="query",
+                default=False,
+            ),
+            OpenApiParameter(
+                name="sample",
+                description="Return events in pseudo-random order. This is deterministic so an identical query will always return the same events in the same order.",
```
Comment (sentaur-athena) [inline src/sentry/issues/endpoints/project_events.py:57]: Visible in docs

File: src/sentry/issues/endpoints/project_events.py [hunk 6]
```diff
@@ -31,26 +38,42 @@ class ProjectEventsEndpoint(ProjectEndpoint):
...
+            404: RESPONSE_NOT_FOUND,
+        },
+        examples=EventExamples.PROJECT_EVENTS_SIMPLE,
+    )
     def get(self, request: Request, project: Project) -> Response:
         """
-        List a Project's Error Events
-        ```````````````````````
-
         Return a list of events bound to a project.
```
Comment (sentaur-athena) [inline src/sentry/issues/endpoints/project_events.py:76]: Visible in docs

File: src/sentry/issues/endpoints/project_events.py [hunk 7]
```diff
@@ -31,26 +38,42 @@ class ProjectEventsEndpoint(ProjectEndpoint):
         }
     }
 
+    @extend_schema(
+        operation_id="List a Project's Error Events",
```
Comment (sentaur-athena) [inline src/sentry/issues/endpoints/project_events.py:42]: Visible in docs

**Review by sentaur-athena — COMMENTED** [review]
Added some comments to help with docs review.

**Review by sentaur-athena — APPROVED** [review]
Looks good to me. Added docs team to see if we're good with the wording.

File: src/sentry/issues/endpoints/project_events.py (hunk 4 above)
Comment (lizokm) [inline src/sentry/issues/endpoints/project_events.py:49]:
```suggestion
                description="If this is set to true, the event payload will include the full event body, including the stacktrace. Set to 1 to enable.",
```

**Review by lizokm — APPROVED** [review]
Looks good from a words perspective. Just one small tweak.

**Pull request merged**